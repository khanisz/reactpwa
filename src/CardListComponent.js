import React, { Component } from "react";
import CardComponent from './CardComponent';


class CardListComponent extends Component {
    constructor(props) {
        super(props);
        const {components} = props;
        const cards = components.map(() =>
            <CardComponent>

            </CardComponent>
        );
    }
    render() {
      return (
        {cards}
      );
    }
  }
  
  export default CardListComponent;