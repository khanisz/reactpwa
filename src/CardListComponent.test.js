import React from "react";
import { shallow } from "enzyme";
import CardListComponent from "./CardListComponent";

describe("CardListComponent", () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<CardListComponent debug />);
    expect(component).toMatchSnapshot();
  });

  it("should render correctly with no props", () => {
    const component = shallow(<CardListComponent />);
    expect(component).toMatchSnapshot();
  });

  it("should render correctly with 0 components", () => {
    const components = [];
    const component = shallow(<CardListComponent components={components} />);
    expect(component).toMatchSnapshot();
  });

  it("should render correctly with 1 components", () => {
    const components = [
      {
        position: 1,
        type: "textC",
        title: "Morderstwo",
        text: "Breslau, 1938 rok. Przedmieście. Brutalnie zamordowano oficera.",
        visible: true
      }
    ];
    const component = shallow(<CardListComponent components={components} />);
    expect(component).toMatchSnapshot();
  });

  it("should render correctly with 2 components", () => {
    const components = [
        {
            "position": 1,
            "type": "textC",
            "title": "Morderstwo",
            "text": "Breslau, 1938 rok. Przedmieście. Brutalnie zamordowano oficera.",
            "visible" : true
        },
        {
            "position": 2,
            "type": "imageC",
            "title": "Kawiarnia",
            "url" : "https://media-cdn.tripadvisor.com/media/photo-s/09/54/12/8e/old-cafe.jpg",
            "text": "W pobliskiej kawiarni czeka żona zmarłego. Posiada ona kluczowe informacje o morderstwie.",
            "visible" : true
        }
    ];
    const component = shallow(<CardListComponent components={components} />);
    expect(component).toMatchSnapshot();
  });

  it("should render correctly with 3 components", () => {
    const components = [        {
        "position": 1,
        "type": "textC",
        "title": "Morderstwo",
        "text": "Breslau, 1938 rok. Przedmieście. Brutalnie zamordowano oficera.",
        "visible" : true
    },
    {
        "position": 2,
        "type": "imageC",
        "title": "Kawiarnia",
        "url" : "https://media-cdn.tripadvisor.com/media/photo-s/09/54/12/8e/old-cafe.jpg",
        "text": "W pobliskiej kawiarni czeka żona zmarłego. Posiada ona kluczowe informacje o morderstwie.",
        "visible" : true
    },
    {
        "position": 3,
        "type": "imageC",
        "title": "Miejsce zbrodni",
        "text": "Na miejscu widać ofiarę leżącą w pokoju. To najpewniej liczne rany kłute doprowadziły ją do tego stanu.",
        "url": "https://www.history.com/.image/ar_16:9%2Cc_fill%2Ccs_srgb%2Cfl_progressive%2Cg_faces:center%2Cq_auto:good%2Cw_768/MTU5MTY4OTAyMTUxMDIyMjI5/5_dp263790.jpg",
        "visible" : false
    }];
    const component = shallow(<CardListComponent components={components} />);
    expect(component).toMatchSnapshot();
  });

  it("should render correctly with 4 different components", () => {
    const components = [        {
        "position": 1,
        "type": "textC",
        "title": "Morderstwo",
        "text": "Breslau, 1938 rok. Przedmieście. Brutalnie zamordowano oficera.",
        "visible" : true
    },
    {
        "position": 2,
        "type": "imageC",
        "title": "Kawiarnia",
        "url" : "https://media-cdn.tripadvisor.com/media/photo-s/09/54/12/8e/old-cafe.jpg",
        "text": "W pobliskiej kawiarni czeka żona zmarłego. Posiada ona kluczowe informacje o morderstwie.",
        "visible" : true
    },
    {
        "position": 3,
        "type": "imageC",
        "title": "Miejsce zbrodni",
        "text": "Na miejscu widać ofiarę leżącą w pokoju. To najpewniej liczne rany kłute doprowadziły ją do tego stanu.",
        "url": "https://www.history.com/.image/ar_16:9%2Cc_fill%2Ccs_srgb%2Cfl_progressive%2Cg_faces:center%2Cq_auto:good%2Cw_768/MTU5MTY4OTAyMTUxMDIyMjI5/5_dp263790.jpg",
        "visible" : false
    },
    {
        "position": 4,
        "type": "questionC",
        "title": "Narzędzie zbrodni",
        "question": "Jakiej broni użyto?",
        "answer": "sztylet",
        "resolution": "Tylko Anna oraz Tomasz mieli dostęp do narzędzia zbrodni.",
        "visible" : false
    }];
    const component = shallow(<CardListComponent components={components} />);
    expect(component).toMatchSnapshot();
  });
});
