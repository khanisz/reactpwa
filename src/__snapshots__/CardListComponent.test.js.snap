// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardListComponent should render correctly in "debug" mode 1`] = `
<div>
  ...Loading
</div>
`;

exports[`CardListComponent should render correctly with 0 components 1`] = `<div />`;

exports[`CardListComponent should render correctly with 1 components 1`] = `
<div>
  <CardComponent
    component={
      Object {
        "position": 1,
        "text": "Breslau, 1938 rok. Przedmieście. Brutalnie zamordowano oficera.",
        "title": "Morderstwo",
        "type": "textC",
        "visible": true,
      }
    }
  />
</div>
`;

exports[`CardListComponent should render correctly with 2 components 1`] = `
<div>
  <CardComponent
    component={
      Object {
        "position": 1,
        "text": "Breslau, 1938 rok. Przedmieście. Brutalnie zamordowano oficera.",
        "title": "Morderstwo",
        "type": "textC",
        "visible": true,
      }
    }
  />
  <CardComponent
    component={
      Object {
        "position": 2,
        "text": "W pobliskiej kawiarni czeka żona zmarłego. Posiada ona kluczowe informacje o morderstwie.",
        "title": "Kawiarnia",
        "type": "imageC",
        "url": "https://media-cdn.tripadvisor.com/media/photo-s/09/54/12/8e/old-cafe.jpg",
        "visible": true,
      }
    }
  />
</div>
`;

exports[`CardListComponent should render correctly with 3 components 1`] = `
<div>
  <CardComponent
    component={
      Object {
        "position": 1,
        "text": "Breslau, 1938 rok. Przedmieście. Brutalnie zamordowano oficera.",
        "title": "Morderstwo",
        "type": "textC",
        "visible": true,
      }
    }
  />
  <CardComponent
    component={
      Object {
        "position": 2,
        "text": "W pobliskiej kawiarni czeka żona zmarłego. Posiada ona kluczowe informacje o morderstwie.",
        "title": "Kawiarnia",
        "type": "imageC",
        "url": "https://media-cdn.tripadvisor.com/media/photo-s/09/54/12/8e/old-cafe.jpg",
        "visible": true,
      }
    }
  />
  <CardComponent
    component={
      Object {
        "position": 3,
        "text": "Na miejscu widać ofiarę leżącą w pokoju. To najpewniej liczne rany kłute doprowadziły ją do tego stanu.",
        "title": "Miejsce zbrodni",
        "type": "imageC",
        "url": "https://www.history.com/.image/ar_16:9%2Cc_fill%2Ccs_srgb%2Cfl_progressive%2Cg_faces:center%2Cq_auto:good%2Cw_768/MTU5MTY4OTAyMTUxMDIyMjI5/5_dp263790.jpg",
        "visible": false,
      }
    }
  />
</div>
`;

exports[`CardListComponent should render correctly with 4 different components 1`] = `
<div>
  <CardComponent
    component={
      Object {
        "position": 1,
        "text": "Breslau, 1938 rok. Przedmieście. Brutalnie zamordowano oficera.",
        "title": "Morderstwo",
        "type": "textC",
        "visible": true,
      }
    }
  />
  <CardComponent
    component={
      Object {
        "position": 2,
        "text": "W pobliskiej kawiarni czeka żona zmarłego. Posiada ona kluczowe informacje o morderstwie.",
        "title": "Kawiarnia",
        "type": "imageC",
        "url": "https://media-cdn.tripadvisor.com/media/photo-s/09/54/12/8e/old-cafe.jpg",
        "visible": true,
      }
    }
  />
  <CardComponent
    component={
      Object {
        "position": 3,
        "text": "Na miejscu widać ofiarę leżącą w pokoju. To najpewniej liczne rany kłute doprowadziły ją do tego stanu.",
        "title": "Miejsce zbrodni",
        "type": "imageC",
        "url": "https://www.history.com/.image/ar_16:9%2Cc_fill%2Ccs_srgb%2Cfl_progressive%2Cg_faces:center%2Cq_auto:good%2Cw_768/MTU5MTY4OTAyMTUxMDIyMjI5/5_dp263790.jpg",
        "visible": false,
      }
    }
  />
  <CardComponent
    component={
      Object {
        "answer": "sztylet",
        "position": 4,
        "question": "Jakiej broni użyto?",
        "resolution": "Tylko Anna oraz Tomasz mieli dostęp do narzędzia zbrodni.",
        "title": "Narzędzie zbrodni",
        "type": "questionC",
        "visible": false,
      }
    }
  />
</div>
`;

exports[`CardListComponent should render correctly with no props 1`] = `
<div>
  ...Loading
</div>
`;
